packages/apollo-angular/src/fragment.ts

```diff
@@ -0,0 +1,16 @@
+import type { Observable } from 'rxjs';
+import { Injectable } from '@angular/core';
+import type { WatchFragmentOptions, WatchFragmentResult } from '@apollo/client';
+import { Apollo } from './apollo';
+import type { EmptyObject } from './types';
+
+@Injectable()
+export class Fragment<T = {}, V = EmptyObject> {
+  public client = 'default';
+
+  constructor(protected apollo: Apollo) {}
+
+  public watchFragment(options?: WatchFragmentOptions<T, V>): Observable<WatchFragmentResult<T>> {
```

Comment:
```suggestion
  public watchFragment(options?: WatchFragmentOptionsAlone<T, V>): Observable<WatchFragmentResult<T>> {
```

This should use a new type `WatchFragmentOptionsAlone` that exclude `fragment` and `from`. And those tow required options should instead be delcared as inheritable members, as is done for `document` in the equivalent `Query` class.

But all of this really makes sense if are able to generate code for fragments too. Is is something you planned ?

Reply:
Thanks @PowerKiKi - I've made part of that change in https://github.com/kamilkisiela/apollo-angular/pull/2226/commits/5978d22deeb7baa6d3547a0f1d5ae736eb3eda79, though after thinking about it a bit more I think it makes sense to exclude `fragment` but not `from`. I imagine that a Fragment service would be created once per fragment document, but that you may want to pass in the object reference via `from` at the call site. An example might be helpful here - would it be possible to create a one-off release from this PR?

Also re: the codegen story for this feature, I was discussing this with @phryneas a bit and he had some ideas around an e.g. `Fragment.from(document)` first-class API instead. Should I create a separate issue for that discussion?

Reply:
It seems I forgot to reply earlier. Sorry about that.

Keeping `from` as an option makes a lot of sense 👍

I'm not sure what the hypothetical `Fragment.from(document)` means. Feel free to explain more here, or in a new thread if you feel like the discussion might become too big/unrelated.